### DISCUSSION TITLE
Testing Components with Lazy loading

### DISCUSSION THREAD

**Discussion by amshehzad** [livewire/livewire#7940] [1. Help]
I'm trying to test the component using Lazy loading but the component does not render, I'm currently calling refresh method in order to render it. couldn't find anything about this in documentation. is there a way to render lazy loaded copmonent in testing?

this is how I'm doing currently in order to render the component:

```
Livewire::test(DocumentViewer::class, ['document' => $document])
    ->call('$refresh')
    ->assertDispatched('flashNotification', message: 'Failed to open the document', type: 'error');
```

it won't render withou refresh

**Comment by joshhanley** [comment]
@amshehzad currently there is no built in way to tell a lazy component to load, so the `$refresh` seems fine to me if it works. PRs welcome if you think there could be a better helper method (that maybe just calls `$refresh` under the hood) 🙂
